packages/typespec-client-generator-core/src/package.ts

```diff
@@ -536,7 +536,7 @@ function getEndpointTypeFromSingleServer<
       }
       const apiVersionInfo = updateWithApiVersionInformation(context, param, client.__raw.type);
       sdkParam.isApiVersionParam = apiVersionInfo.isApiVersionParam;
-      if (sdkParam.isApiVersionParam) {
+      if (sdkParam.isApiVersionParam && apiVersionInfo.clientDefaultValue) {
```

Comment:
Is the `sdkParam.clientDefaultValue` always be override if apiVersionInfo.clientDefaultValue (if exist)? Is it possible that the sdkParam.clientDefaultValue has different value of  apiVersionInfo.clientDefaultValue, in that way , the sdkParam itself  default value is lost.

Considering the versioned service, and the `apiVersion` in the URL, with a default value (which may or may not the any enum value of the version enum)
Do we have UT to test this scenario?

Reply:
for versioning service, we honor the api version emitter specified, not client default value.